### ISSUE TITLE
Failing in adding a Universe creates bad state

### ISSUE THREAD

**Issue by richardjgowers** [datreant/MDSynthesis#44]
```
In [1]: import mdsynthesis as mds

In [2]: S = mds.Sim('cg')

In [5]: S.universes.add('main', ['topol.tpr','cg.xtc'])
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-5-f7469218f877> in <module>()
----> 1 S.universes.add('main', ['topol.tpr','cg.xtc'])

/home/richard/.local/lib/python2.7/site-packages/mdsynthesis-0.5.0-py2.7.egg/mdsynthesis/core/aggregators.py in add(self, handle, topology, *trajectory)
    333                 outtraj.append(traj)
    334 
--> 335         self._backend.add_universe(handle, topology, *outtraj)
    336 
    337         if not self.default():

/home/richard/.local/lib/python2.7/site-packages/mdsynthesis-0.5.0-py2.7.egg/mdsynthesis/core/persistence.py in inner(self, *args, **kwargs)
    224             self._exlock(self.handle)
    225             try:
--> 226                 out = func(self, *args, **kwargs)
    227             finally:
    228                 self.handle.close()

/home/richard/.local/lib/python2.7/site-packages/mdsynthesis-0.5.0-py2.7.egg/mdsynthesis/core/persistence.py in add_universe(self, universe, topology, *trajectory)
    880 
    881         # add topology paths to table
--> 882         table.row['abspath'] = os.path.abspath(topology)
    883         table.row['relCont'] = os.path.relpath(topology, self.get_location())
    884         table.row.append()

/usr/lib/python2.7/posixpath.pyc in abspath(path)
    365 def abspath(path):
    366     """Return an absolute path."""
--> 367     if not isabs(path):
    368         if isinstance(path, _unicode):
    369             cwd = os.getcwdu()

/usr/lib/python2.7/posixpath.pyc in isabs(s)
     59 def isabs(s):
     60     """Test whether a path is absolute"""
---> 61     return s.startswith('/')
     62 
     63 

AttributeError: 'list' object has no attribute 'startswith'

In [6]: S.universes.add('main', 'topol.tpr','cg.xtc')
---------------------------------------------------------------------------
NoSuchNodeError                           Traceback (most recent call last)
<ipython-input-6-f176da1cef98> in <module>()
----> 1 S.universes.add('main', 'topol.tpr','cg.xtc')

/home/richard/.local/lib/python2.7/site-packages/mdsynthesis-0.5.0-py2.7.egg/mdsynthesis/core/aggregators.py in add(self, handle, topology, *trajectory)
    333                 outtraj.append(traj)
    334 
--> 335         self._backend.add_universe(handle, topology, *outtraj)
    336 
    337         if not self.default():

/home/richard/.local/lib/python2.7/site-packages/mdsynthesis-0.5.0-py2.7.egg/mdsynthesis/core/persistence.py in inner(self, *args, **kwargs)
    224             self._exlock(self.handle)
    225             try:
--> 226                 out = func(self, *args, **kwargs)
    227             finally:
    228                 self.handle.close()

/home/richard/.local/lib/python2.7/site-packages/mdsynthesis-0.5.0-py2.7.egg/mdsynthesis/core/persistence.py in add_universe(self, universe, topology, *trajectory)
    872                 '/universes/{}'.format(universe), 'topology')
    873             self.handle.remove_node(
--> 874                 '/universes/{}'.format(universe), 'trajectory')
    875 
    876         # construct topology table

/home/richard/.local/lib/python2.7/site-packages/tables/file.pyc in remove_node(self, where, name, recursive)
   1779         """
   1780 
-> 1781         obj = self.get_node(where, name=name)
   1782         obj._f_remove(recursive)
   1783 

/home/richard/.local/lib/python2.7/site-packages/tables/file.pyc in get_node(self, where, name, classname)
   1614         # Now we have the definitive node path, let us try to get the node.
   1615         if node is None:
-> 1616             node = self._get_node(nodepath)
   1617 
   1618         # Finally, check whether the desired node is an instance

/home/richard/.local/lib/python2.7/site-packages/tables/file.pyc in _get_node(self, nodepath)
   1553             return self.root
   1554 
-> 1555         node = self._node_manager.get_node(nodepath)
   1556         assert node is not None, "unable to instantiate node ``%s``" % nodepath
   1557 

/home/richard/.local/lib/python2.7/site-packages/tables/file.pyc in get_node(self, key)
    434 
    435         if self.node_factory:
--> 436             node = self.node_factory(key)
    437             self.cache_node(node, key)
    438 

/home/richard/.local/lib/python2.7/site-packages/tables/group.pyc in _g_load_child(self, childname)
   1184             childname = join_path(self._v_file.root_uep, childname)
   1185         # Is the node a group or a leaf?
-> 1186         node_type = self._g_check_has_child(childname)
   1187 
   1188         # Nodes that HDF5 report as H5G_UNKNOWN

/home/richard/.local/lib/python2.7/site-packages/tables/group.pyc in _g_check_has_child(self, name)
    400             raise NoSuchNodeError(
    401                 "group ``%s`` does not have a child named ``%s``"
--> 402                 % (self._v_pathname, name))
    403         return node_type
    404 

NoSuchNodeError: group ``/`` does not have a child named ``/universes/main/trajectory``
```

**Comment by dotsdl** [comment]
This will be solved in #45.

**Comment by dotsdl** [comment]
Closing, since we now use JSON for storage, so no more HDF5 issues.
